src/hotspot/share/jfr/leakprofiler/checkpoint/objectSampleWriter.cpp

```diff
@@ -199,7 +199,7 @@ static ArrayInfo* array_infos = nullptr;
 static FieldTable* field_infos = nullptr;
 static RootDescriptionInfo* root_infos = nullptr;
 
-int __write_sample_info__(JfrCheckpointWriter* writer, const void* si) {
+static int __write_sample_info__(JfrCheckpointWriter* writer, const void* si) {
```

Comment:
Please feel free to open a bug report. 😉 

Unless there is a warning flag to avoid creating reserved names (is there?), it is more of a matter of coding style on the part of Hotspot, and that is basically where I draw the line of my meddling with Hotspot. :)

Reply:
I just discovered that Clang 13 added `-Wreserved-identifier`.  There's also an open gcc bug:
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=51437

And a discussion of the "chattiness" of the feature:
https://github.com/llvm/llvm-project/issues/57913#issuecomment-1255493025

Probably there's not much appetite for this sort of thing, and I shouldn't have brought it up.